extensions/lifecycle/tasks/LifecycleColdStatusArchiveTask.js

```diff
@@ -0,0 +1,147 @@
+const async = require('async');
+const { errors } = require('arsenal');
+
+const ObjectMDArchive = require('arsenal').models.ObjectMDArchive;
+const LifecycleUpdateTransitionTask = require('./LifecycleUpdateTransitionTask');
+
+class LifecycleColdStatusArchiveTask extends LifecycleUpdateTransitionTask {
+    /**
+     * Process a lifecycle object entry
+     *
+     * @constructor
+     * @param {LifecycleObjectProcessor} proc - object processor instance
+     */
+    constructor(proc) {
+        const procState = proc.getStateVars();
+        super();
+        Object.assign(this, procState);
+    }
+
+    getTargetAttribute(entry) {
+        const {
+            bucketName,
+            objectKey,
+            objectVersion,
+            accountId
+        } = entry.getTarget();
+        return {
+            bucket: bucketName,
+            key: objectKey,
+            version: objectVersion,
+            accountId
+        };
+    }
+
+    getSourceAttribute(entry) {
+        const { bucketName, objectKey, objectVersion } = entry.getTarget();
+        return {
+            bucket: bucketName,
+            objectKey,
+            version: objectVersion,
+            storageClass: entry.get('sourceStorageClass'),
+            lastModified: entry.get('sourceLastModified'),
+        };
+    }
+
+    /**
+     * delete data in set in ObjectMD location
+     * @param {object} entry - ColdStorageActionQueueEntry
```

Comment:
I am not sure if JSDoc will properly handle the variable type but at least until we switch to type script this should be close enough.
```suggestion
     * @param {ColdStorageActionQueueEntry} entry - what entry to delete
```
( feel free to fix my description :smirk: )